ssl.h

```diff
@@ -22,25 +23,26 @@
 #ifndef _RDSSL_H
 #define _RDSSL_H
 
-#include <openssl/rc4.h>
-#include <openssl/md5.h>
-#include <openssl/sha.h>
-#include <openssl/err.h>
-#include <openssl/bn.h>
-#include <openssl/x509v3.h>
-#include <openssl/hmac.h>
+#include <gnutls/x509.h>
+#include <gnutls/abstract.h>
 
-#if defined(OPENSSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x0090800f)
-#define D2I_X509_CONST const
-#else
-#define D2I_X509_CONST
-#endif
+#include <nettle/md5.h>
+#include <nettle/sha1.h>
+#include <nettle/arcfour.h>
+#include <nettle/hmac.h>
+#include <nettle/rsa.h>
+
+#include <gmp.h>
+#include <libtasn1.h>
```

Comment:
Do we really need all of these headers here? It's good to keep the dependency chains as small as possible.